ui/apps/platform/src/Containers/Collections/hooks/useDryRunConfiguration.ts

```diff
@@ -28,11 +28,12 @@ export default function useDryRunConfiguration(
                 dryRunConfig.resourceSelectors,
                 nextConfig.resourceSelectors
             );
-            if (isEmbeddedCollectionsChanged || isResourceSelectorChanged) {
+            const isNameChange = dryRunConfig.name !== nextConfig.name;
+            if (isEmbeddedCollectionsChanged || isResourceSelectorChanged || isNameChange) {
```

Comment:
It might seem like overkill, but we need to send dry run requests on name change, since name is a part of the server side validation. This will force users to enter _some_ correct name value before previewing deployments via rules.